### PR TITLE
Add once event handler

### PR DESCRIPTION
**Pull request by rscardinho**
This PR adds an `#once` emitter, so an event can be added, emitted and removed from listeners list with a single command

### CODE REVIEW

File: spec/emittr/callback_spec.rb [hunk 1]
```diff
@@ -21,32 +21,67 @@
...
       end
     end
 
-    context 'when a different callback is passed' do
-      it 'returns false' do
-        wrapper_callback = Proc.new {}
-        unset_callback = Proc.new {}
+    describe 'when an Emittr::Callback instance is provided' do
+      context 'when same callbacks' do
+        it 'returns true' do
```
Comment (talyssonoc) [inline spec/emittr/callback_spec.rb]: Can we use `it 'should ...'` in those test cases?

Comment (rscardinho) [inline spec/emittr/callback_spec.rb]: its a best practice for rspec http://betterspecs.org/#should ;)

File: lib/emittr/events.rb [hunk 2]
```diff
@@ -5,25 +5,40 @@ def self.included(klass)
...
           listeners.delete event
         end
 
         self
       end
 
+      def once(event, &block)
+        callback = ::Emittr::Callback.new &block
+
+        off_block = Proc.new do |args|
```
Comment (talyssonoc) [inline lib/emittr/events.rb:30]: 👏

File: spec/emittr_spec.rb [hunk 3]
```diff
@@ -77,6 +77,29 @@
     end
   end
 
+  describe '#once' do
+    let(:callback) { Proc.new {} }
+
+    before { emitter.once :once_test, &callback }
+
+    it 'calls callback' do
```
Comment (talyssonoc) [inline spec/emittr_spec.rb]: Maybe we could add a `context` here? Like `context 'when event is triggered'`.

We should also test if the callback is called only once. What about  putting an `once` on the `receive` and emit twice to make sure it's emitted only once?

File: spec/emittr_spec.rb [hunk 4]
```diff
@@ -77,6 +77,29 @@
...
+      expect(callback).to receive(:call)
+      emitter.emit :once_test
+    end
+
+    it 'adds listeners to list when not emitted yet' do
+      listeners = emitter.send(:listeners)
+      expect(listeners[:once_test]).to be_any
```
Comment (talyssonoc) [inline spec/emittr_spec.rb]: I'm not sure I understood that test. Could you elaborate?

Comment (rscardinho) [inline spec/emittr_spec.rb]: yeah, the `be_any` matcher makes it very "unclear". I'll think in a better expectation for this

File: spec/emittr_spec.rb (hunk 3 above)
Comment (rscardinho) [inline spec/emittr_spec.rb]: yup, I'll rewrite the spec to test what `#once` should do

**Pull request merged**